src/ln/peer_handler.rs

```diff
@@ -151,12 +155,25 @@ impl<Descriptor: SocketDescriptor> PeerHolder<Descriptor> {
 	}
 }
 
+#[cfg(all(target_pointer_width = "32", target_pointer_width = "64"))]
```

Comment:
1. This cfg is meaningless, you'll never have a pointer which is both 32 and 64 bits.
2. `_check_usize_is_64` isn't exactly right because you check `*const usize` not usize, so you actually does check the size of the pointer.
3. in 32bits this should be `[u8; 4]` but in 64bits this should be `[u8; 8]`

Reply:
Oops, duh, thanks, that was for testing. For 3, its right, cause we just have to use either 4 or 8 so that it fails in all cases where pointer isnt 4 or 8.

Reply:
oh you're doing `not(all` kk